[5.7] Fixed situation with overridden base Application-class

mingalevme opened the pull request:
My Application-class (**App\Helpers\Laravel\Lumen\Application**) override base-class (**Laravel\Lumen\Application**). The overriding was made in the middle of project life so part of the code expects **Laravel\Lumen\Application** and part - **App\Helpers\Laravel\Lumen\Application**.

GrahamCampbell commented:
What's your use case for extending the application class?

mingalevme commented:
```
public function isDownForMaintenance()
    {
        $filename = app()->storagePath() . '/lib/mfeed/down';
        return file_exists($filename);
    }
```
and

```
public function call($callback, array $parameters = [], $defaultMethod = null)
    {
        if (is_array($callback) && $callback[0] instanceof Command && $callback[1] === 'handle') {
            \Log::debug("Command is executing", [
                'command' => get_class($callback[0]),
            ]);
            $response = \Event::fire(new CommandExecuting($callback[0], $parameters), [], true);
            if ($response !== null) {
                \Log::debug("Execution of the command has been halted by a listener", [
                    'command' => get_class($callback[0]),
                    'response' => $response,
                ]);
                return null;
            }
        }

        $start = microtime(true);
        $result = parent::call($callback, $parameters, $defaultMethod);
        $time = microtime(true) - $start;

        if (is_array($callback) && $callback[0] instanceof \App\Jobs\BaseJob && $callback[1] === 'handle') {
            \Event::fire(new JobProcessed($callback[0]));
            \Log::debug("Job has been processed", [
                'job' => get_class($callback[0]),
                'time' => round($time, 3),
            ]);
        }

        if (is_array($callback) && $callback[0] instanceof Command && $callback[1] === 'handle') {
            \Event::fire(new CommandExecuted($callback[0], $parameters));
            \Log::debug("Command has been executed", [
                'command' => get_class($callback[0]),
            ]);
        }
        
        return $result;
    }
```

driesvints commented:
@mingalevme I don't understand why you'd use two separate `Application` classes. You either you the one provided by Lumen or overwrite it with your custom one.

Btw you `isDownForMaintenance` has an `app()` call which can be replaced by `$this`.

mingalevme commented:
I've fixed it by:
```$app->alias(App\Helpers\Laravel\Lumen\Application::class, \Laravel\Lumen\Application::class);```
but I think it may be useful for someone else.

driesvints commented:
> I've fixed it by:
> `$app->alias(App\Helpers\Laravel\Lumen\Application::class, \Laravel\Lumen\Application::class);`
> but I think it may be useful for someone else.

You should only replace `$app = new Laravel\Lumen\Application(` in your `bootstrap/app.php` file with `$app = new App\Helpers\Laravel\Lumen\Application(` really.

driesvints commented:
@mingalevme also: can you check if you're on 5.7.6?

```bash
php artisan version
```

mingalevme commented:
> @mingalevme I don't understand why you'd use two separate `Application` classes. You either you the one provided by Lumen or overwrite it with your custom one.
> 
I use only one Application-class - my custom, I added it not in the begging of the project's life, but when I added it I didn't replace all occurrences of ```Laravel\Lumen\Application```, and that worked in 5.6-. Now I've upgraded to 5.7 and that has been broken.

mingalevme commented:
> $app = new App\Helpers\Laravel\Lumen\Application(...)
> 
That I did when override the class.

mingalevme commented:
The biggest problem with it - that it take a lot of time to understand what exactly has been breaking after upgrading to 5.7.

mingalevme commented:
> @mingalevme also: can you check if you're on 5.7.6?
> 
> ```shell
> php artisan version
> ```

Lumen has no ```version``` command

driesvints commented:
@mingalevme if you upgrade to 5.7.6 this should be fixed.

driesvints commented:
See https://github.com/laravel/lumen-framework/issues/849

driesvints commented:
@mingalevme why would you expect the base class if you're overwriting it? The `bootstrap/app.php` has the only occurrence.

mingalevme commented:
Because there are legacy classes, I didn't want to touch them to didn't break something.
